### PR TITLE
feat(EventTracker): Use sendBeacon instead of Fetch with dispatch on unload rather than on next load.

### LINKED ISSUE #161: Use The Web Beacon API for EventsKit

### PR DESCRIPTION
**Pull request by klarstrup**
Closes #161

### CODE REVIEW

File: lib/kits/events/tracker.ts [hunk 1]
```diff
@@ -239,14 +242,14 @@ class Tracker {
         };
 
         if (this.eventsTrackUrl) {
-            dispatch(this.eventsTrackUrl, this.eventsTrackHeaders);
+            this.dispatch();
             this.hasMadeInitialDispatch = true;
```
Comment (klarstrup) [inline lib/kits/events/tracker.ts:299]: This on-instantiation dispatching and tracking of `hasMadeInitialDispatch` we can also get rid of if there aren't events(from a persisted pool) to dispatch immediately when `new Tracker`, but we would want to cover that for a transition for any backed-up event pool for a week or whatever

**Comment by mortenbo** [comment]
@klarstrup I think it looks really good. Let's start out by testing this in Vizsla staging before we release this anywhere else.

**Review by mortenbo — APPROVED** [review]

**Comment by klarstrup** [comment]
Then I'd like to see you and @jrlarano merge your approved PRs so those can be released and this can be moved ahead into beta for Vizsla's usage

**Comment by mortenbo** [comment]
I've deleted my PR as it's not relevant after all 👍

**Review by jrlarano — APPROVED** [review]

File: lib/kits/events/tracker.ts [hunk 2]
```diff
@@ -351,70 +394,16 @@ class Tracker {
...
-    async (
-        eventsTrackUrl: string,
-        eventsTrackHeaders: Record<string, string>
-    ) => {
-        if (!pool) {
-            console.warn('Tracker: dispatch called with no active event pool.');
-            return;
+    dispatching = false;
+    dispatchLimit = 100;
+    dispatch = throttle(this.dispatchBeacon, 4000);
```
Comment (klarstrup) [inline lib/kits/events/tracker.ts]: Going to swap this `throttle` for a more suitable `debounce`, such that we don't do a lot of immediate single-event dispatches.

**Pull request merged**